### ISSUE TITLE
[Bug Report]: Client side error on dev build when devServer.hot is set to true

### ISSUE THREAD

**Issue by zakuru** [web-infra-dev/rspack#2526]
### System Info

  System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz
    Memory: 38.27 GB / 63.73 GB
  Binaries:
    Node: 14.20.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files\nodejs\yarn.CMD
    npm: 6.14.17 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 111.0.5563.147
    Edge: Spartan (44.19041.1266.0), Chromium (111.0.1661.51)
    Internet Explorer: 11.0.19041.1566

### Details

![image](https://user-images.githubusercontent.com/171174/228457631-cdc8c0d7-9e1b-48f6-ba12-61c0f12e51ad.png)

```
Uncaught TypeError: Cannot read properties of undefined (reading 'createSignatureFunctionForTransform')
    at ./src/App.jsx (src\App.jsx:12:12)
    at __webpack_require__ (src\index.jsx:9:9)
    at fn (src\index.jsx:9:9)
    at ./src/index.jsx (src\App.jsx:9:5)
    at __webpack_require__ (src\index.jsx:9:9)
    at src\index.jsx:9:9
    at src\index.jsx:9:9
./src/App.jsx @ src\App.jsx:12
__webpack_require__ @ src\index.jsx:9
fn @ src\index.jsx:9
./src/index.jsx @ src\App.jsx:9
__webpack_require__ @ src\index.jsx:9
(anonymous) @ src\index.jsx:9
(anonymous) @ src\index.jsx:9
```

the issue can be worked around by setting the hot option to `false`

### Reproduce Steps

Go into the react example folder, mitigate issue https://github.com/web-infra-dev/rspack/issues/2525 by setting `devServer.webSocketServer` to `false` then run yarn dev 
The build will succeed but the page wont load because of the client side error.

The issue can be worked around by setting `devServer.hot` to `false`

**Comment by hyf0** [comment]
workaround: you should be able to avoid this problem by setting `devServer.hot: false`.

**Comment by zakuru** [comment]
@IWANABETHATGUY  @hyf0  I will close this issue as it is fixed in version 0.1.6
